### ISSUE TITLE
Arabic calendar has one day difference from the actual date

### ISSUE THREAD

**Issue by mnrafg** [fisharebest/ext-calendar#9]
```
// Set Timezone
date_default_timezone_set('Asia/Riyadh');

// Today
list($year, $month, $day) = explode(',', date('Y,m,d'), time());

// Create Calendars
$gc = new GregorianCalendar;
$ac = new ArabicCalendar;

// Julian Day
$jd = $gc->ymdToJd($year, $month, $day); // 2017, 4, 5

// Convert to Arabic Date
$ad = $ac->jdToYmd($julian_day); // [1438, 7, 9]

// NOTE: [1438, 7, 9] is not correct because it actually is [1438, 7, 8] today.
```

You can check today's date by using the code above and visit the following Saudi Arabia government official website to compare the dates.
[http://www.mofa.gov.sa/sites/mofaen](http://www.mofa.gov.sa/sites/mofaen)
**Note:** today's confidential Arabic date is available at the top left corner of the above website.

The following website is also another confidential source of Arabic calendars, and it also shows the same date as the Suadi Arabia website is pointing to.
[https://www.islamicfinder.org/?language=en](https://www.islamicfinder.org/?language=en)

In my case I mean for today the first website shows the following date:
![date](https://cloud.githubusercontent.com/assets/22196871/24697081/189bc190-1a01-11e7-9155-afd7b8edba5a.png)

And the second website also shows the following date:
![date1](https://cloud.githubusercontent.com/assets/22196871/24697426/55e81b38-1a02-11e7-8bf9-7334e3282bbe.png)

**Comment by fisharebest** [comment]
Let me investigate the algorithm that is used.

**Comment by fisharebest** [comment]
Fixed in 2.3.0

We now use the "civil" calendar, instead of the "astronomical" calendar.
Thanks to @jon48 for the code.
